src/main/kotlin/com/github/gradle/node/npm/task/NpmInstallTask.kt

```diff
@@ -64,8 +64,10 @@ abstract class NpmInstallTask : NpmTask() {
         return projectFileIfExists("yarn.lock").orNull
     }
 
-    @Optional
-    @OutputFile
+    // does this output overlap with getNodeModulesFiles()?
+    //@Optional
+    //@OutputFile
+    @Internal
     protected fun getPackageLockFileAsOutput(): File? {
```

Comment:
Instead of removing the `@Optional @OutputFile` you probably want to follow the pattern below
i.e. add a configurable flag for which behaviour and if it's set to the new behaviour then just return null here immediately

(Though at that point I suspect all the output methods might need to be reworked slightly, but that's boring tedious work and you can happily leave that to the maintainers)